README.md

```diff
@@ -1,64 +1,147 @@
-{2, nil}
-
-iex> Repo.insert_all(Example, [%{a: 5, b: "5"}, %{a: 6}])
-{2, nil}
-
-iex> Example |> order_by(desc: :a) |> limit(2) |> Repo.all()
-[
-  %Example{
-    a: 6,
-    b: "",
-    c: ~N[1970-01-01 00:00:00]
-  }
-  %Example{
-    a: 5,
-    b: "5",
-    c: ~N[1970-01-01 00:00:00]
-  }
+types = [
+  numeric_types_need_size: :u32,
+  no_custom_type_for_strings: :string,
+  datetime: :datetime,
+  maybe_name: {:nullable, :string},
+  country_code: {:string, _size = 2},
+  price: {:decimal, _size = 32, _scale = 2}
 ]
 
-iex> Repo.insert_all(Example, select(Example, [e], %{a: e.a, b: e.b}))
-{4, nil}
+MyApp.Repo.insert_all("example", rows, types: types)
+```
+
+#### Settings
+
+`:settings` option can be used to enable [asynchronous inserts,](https://clickhouse.com/docs/en/optimize/asynchronous-inserts) lightweght [deletes,](https://clickhouse.com/docs/en/guides/developer/lightweght-delete) and [more](https://clickhouse.com/docs/en/operations/settings/settings)
+
+```elixir
+MyApp.Repo.insert_all(MyApp.Example, rows, settings: [async_insert: 1])
+MyApp.Repo.delete_all("example", settings: [allow_experimental_lightweight_delete: 1])
+```
+
+#### UNION and co
+
+As of version `22.8` ClickHouse [doesn't allow](https://clickhouse.com/docs/en/sql-reference/statements/select/union) ordering or limiting combined queries:
+
+> ORDER BY and LIMIT are applied to separate queries, not to the final result. If you need to apply a conversion to the final result, you can put all the queries with UNION in a subquery in the FROM clause.
+
+As a workaround all combination operators are moved to the end of the query. Please open an issue if that causes problems.
 
-iex> Repo.update_all(Example, set: [a: 2])
-# ** (Ecto.QueryError) ClickHouse does not support UPDATE statements -- use ALTER TABLE instead in query:
-# from e0 in Dev.Example,
-#   update: [set: [a: ^...]]
+```elixir
+hits = "hits" |> limit(2) |> order_by(desc: :count) |> select([h], h.counter_id)
+visits = "visits" |> limit(2) |> order_by(desc: :count) |> select([v], v.counter_id)
+
+{sql, _params} = Repo.to_sql(:all, union_all(hits, ^visits))
+
+IO.puts(sql)
+```
+
+(Prettified) output:
 
-# count is 0 since clickhouse doesn't (seem to) respond with how many rows been deleted
-iex> Repo.delete_all(Example, settings: [allow_experimental_lightweight_delete: 1, mutations_sync: 1])
-{0, nil}
+```sql
+SELECT h0."counter_id"
+FROM "hits" AS h0
+ORDER BY h0."count" DESC
+LIMIT 2
+UNION ALL (
+  SELECT v0."counter_id"
+  FROM "visits" AS v0
+  ORDER BY v0."count" DESC
+  LIMIT 2
+)
+```
 
-iex> Repo.aggregate(Example, :count)
-0
+Compare it with the same query generated by Postgres adapter:
 
-iex> Repo.query!("drop table example")
+```sql
+SELECT h0."counter_id"
+FROM "hits" AS h0
+UNION ALL (
+  SELECT v0."counter_id"
+  FROM "visits" AS v0
+  ORDER BY v0."count" DESC
+  LIMIT 2
+)
+ORDER BY h0."count" DESC
+LIMIT 2
```

Comment:
Should we maybe add an example on how to reproduce this same postgresql behaviour in clickhouse? 

Maybe with an additional subquery we could get:
```elixir
subq = union_all(hits, ^visits)
same_behaviour_as_postgresql = from(f in subquery(subq), limit: 2, order_by: [desc: :count]) 
```

```sql
select f."counter_id" from (
  SELECT h0."counter_id"
  FROM "hits" AS h0
  UNION ALL (
    SELECT v0."counter_id"
    FROM "visits" AS v0
    ORDER BY v0."count" DESC
    LIMIT 2
  )
) f
ORDER BY f."count" DESC
LIMIT 2
```

( I did not test my code but believe that should do it 🙈  )

Reply:
https://github.com/plausible/chto/issues/38

Reply:
Very nice! :)